devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1365,6 +1866,579 @@ def deleteCode(codeID) {
 	], 4200)
 }
 
+/**
+ * API end-point for updating the 'lockCodesData' attribute with the user info
+ *
+ * @param userInfo: Map consisting values for code id, phone number, access type, and last usage
+ */
+def setUserInfo(userInfo) {
+	log.trace "[DTH] Executing 'setUserInfo()' by ${device.displayName} with userInfo := ${userInfo}"
+	def codeID = userInfo.codeID
+	if (codeID) {
+		codeID = codeID.toString()
+		def lockCodesData = loadlockCodesData(codeID)
+		lockCodesData[codeID].lastUpdated = userInfo.lastUpdated
+		lockCodesData[codeID].userInfo.phoneNum = userInfo.phoneNum ?: ""
+		def accessType = userInfo.accessType
+		if (accessType) {
+			if (accessType == "always") {
+				lockCodesData[codeID].scheduleInfo = [:]
+				lockCodesData[codeID].scheduleInfo.scheduleData = [:]
+			}
+			lockCodesData[codeID].scheduleInfo.scheduleType = accessType
+		}
+		sendEvent(lockCodesDataEvent(lockCodesData))
+	}
+}
+
+/**
+ * Starts the scanning of schedule for the first code present in the 'lockCodes' map
+ */
+def startScanningSchedule() {
+	log.trace "[DTH] Executing 'startScanningSchedule()' by ${device.displayName}"
+	def codesToScanSchedules = state.codesToScanSchedules
+	def codeID = -1
+	if (codesToScanSchedules && codesToScanSchedules.size() > 0) {
+		// Schedule scanning did not complete the last time. Scan the remaining schedules.
+		codeID = codesToScanSchedules[0]
+	} else {
+		def lockCodes = loadLockCodes()
+		if (lockCodes && lockCodes.size() > 0) {
+			def keySet = lockCodes.keySet()
+			codesToScanSchedules = []
+			// Saving the list of lock codes for which the schedule scanning should be executed
+			// In case the user creates schedule while the scanning is in progress, those codes need not be scanned
+			for (def i = 1; i <= MAX_CODES_TO_SCAN; i++) {
+				if (i.toString() in keySet) {
+					codesToScanSchedules << i
+				}
+			}
+			state.codesToScanSchedules = codesToScanSchedules
+			codeID = codesToScanSchedules[0]
+		}
+	}
+	log.trace "[DTH] startScanningSchedule() -  Schedule to scan := ${state.codesToScanSchedules}"
+	if (codeID != -1) {
+		scanSchedule(codeID)
+	} else {
+		// no schedule to scan - scheduling scanning complete
+		codeScanCompleteEvent()
+	}
+}
+
+/**
+ * Scans the schedule for the specified code slot number
+ * 
+ * @param codeID: The code slot number
+ */
+private def scanSchedule(codeID) {
+	if (isKwiksetLock()) {
+		// For Kwikset locks we can find out the user type and then issue the appropriate schedule get command
+		log.trace "[DTH] scanSchedule() -  Calling configurationGet() for codeID := $codeID"
+		sendHubCommand(new physicalgraph.device.HubAction(secure(zwave.configurationV2.configurationGet(parameterNumber: codeID))))
```

Comment:
I added some code recently so that you shouldn't have to cast Zwave commands to HubActions anymore for `sendHubCommand`